File: 3j/cord/kadai-17/17-22059-来間空-kadai-17.c
#include<stdio.h>

void nisin(unsigned int x){
    unsigned int temp[32];
    for(int i = 0; i < 32; i++){
        temp[31-i] = x%2;
        x/=2;
    }
    for(int i = 0; i < 32; i++){
        printf("%d", temp[i]);
    }
   
   
}

int right_shift(unsigned int r, unsigned int n){
    for(int i = 0; i < n; i++){
        r = r/2;
    }
    return r;
   
}
int left_shift(unsigned int l, unsigned int n){
    for(int i = 0; i < n; i++){
        l = l*2;
    }
    return l;
}

int main(){
    unsigned int a,n = 0;
    printf("非負の整数を入力："); scanf("%u", &a);
    printf("シフトするビット数を入力："); scanf("%u", &n);
    printf("入力した非負の整数（括弧内は2進数）：%u（",  a ); nisin(a);printf("）\n");
    printf("シフト演算子を使って右に%uビットシフト：%u（",n, a>>n); nisin(a>>n);printf("）\n");
    printf("シフト演算子を使わず右に%uビットシフト：",n);printf("%d",right_shift(a,n));printf("（");nisin(right_shift(a,n));printf("）\n");
    printf("シフト演算子を使って左に%uビットシフト：%u（",n, a<<n);nisin(a<<n);printf("）\n");
    printf("シフト演算子を使わず左に%uビットシフト：",n);printf("%d",left_shift(a,n));printf("(");nisin(left_shift(a,n));printf(")\n");
    return 0;
    
}

File: 3j/cord/kadai-17/17-22059-来間空-kadai-17.py
def nisin(x):
    temp = [0] * 32
    for i in range(32):
        temp[31-i] = x % 2
        x //= 2
    for i in range(32):
        print(temp[i], end="")

def main():
    a, b = 0, 0
    a = int(input("正の整数aを入力："))
    b = int(input("負の整数bを入力："))
    print("aの２進数は")
    nisin(a)
    print("bの２進数は")
    nisin(b)
    print("a&bは")
    print(a & b)
    print(nisin(a & b))
    print("a|bは")
    print(a | b)
    print(nisin(a | b))
    print("a^bは")
    print(a ^ b)
    print(nisin(a ^ b))
    print("~aは")
    print(~a)
    print(nisin(~a))
    print("~bは")
    print(~b)
    print(nisin(~b))
    print("bを3ビット右シフトした値は")
    print(b >> 3)
    print(nisin(b >> 3))
    print("bを5ビット左シフトした値は")
    print(b << 5)
    print(nisin(b << 5))

main()